### PR TITLE
syncer: fix panic when concurrent call Status

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
before https://github.com/pingcap/dm/pull/2076, the first print status is happened after a tick (30s), so s.start is very likely initialized. After that PR it's more possible that race happened and further cause a panic.

### What is changed and how it works?
use atomic type and check zero before division.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: syncer/status_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+func (*mockCheckpoint) FlushedGlobalPoint() binlog.Location {
+	return binlog.Location{
+		Position: mysql.Position{
+			Name: "mysql-bin.000123",
+			Pos: 123,
+		},
+	}
+}
```
Comment (Ehco1996) [inline syncer/status_test.go:64]: end line

File: syncer/status_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	wg := sync.WaitGroup{}
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		go func() {
+			ret := s.Status(sourceStatus)
+			status := ret.(*pb.SyncStatus)
+			c.Assert(status.MasterBinlog, Equals, "(mysql-bin.000123, 223)")
+			c.Assert(status.SyncerBinlog, Equals, "(mysql-bin.000123, 123)")
+		}()
+	}
```
Comment (Ehco1996) [inline syncer/status_test.go:50]: do we need wait wg?

**Review by Ehco1996 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**